python/ray/experimental/async_plasma.py

```diff
@@ -1,212 +1,59 @@
-        """Manually cancel all tasks assigned to this event loop."""
-        # Because remove all futures will trigger `set_result`,
-        # we cancel itself first.
-        super().cancel()
-        for future in self.traverse():
-            # All cancelled futures should have callbacks to removed itself
-            # from this linked list. However, these callbacks are scheduled in
-            # an event loop, so we could still find them in our list.
-            if not future.cancelled():
-                future.cancel()
-
-    def set_result(self, result):
-        """Complete all tasks. """
-        for future in self.traverse():
-            # All cancelled futures should have callbacks to removed itself
-            # from this linked list. However, these callbacks are scheduled in
-            # an event loop, so we could still find them in our list.
-            future.set_result(result)
-        if not self.done():
-            super().set_result(result)
-
-    def traverse(self):
-        """Traverse this linked list.
-
-        Yields:
-            PlasmaObjectFuture: PlasmaObjectFuture instances.
-        """
-        current = self.head
-        while current is not None:
-            yield current
-            current = current.next
+    """This class is a wrapper for a Future on Plasma."""
+    pass
 
 
 class PlasmaEventHandler:
     """This class is an event handler for Plasma."""
 
-    def __init__(self, loop, worker):
+    def __init__(self, loop):
         super().__init__()
         self._loop = loop
-        self._worker = worker
-        self._waiting_dict = {}
+        self._waiting_dict = defaultdict(list)
 
     def process_notifications(self, messages):
         """Process notifications."""
         for object_id, object_size, metadata_size in messages:
             if object_size > 0 and object_id in self._waiting_dict:
-                linked_list = self._waiting_dict[object_id]
-                self._complete_future(linked_list)
+                self._complete_future(object_id)
 
     def close(self):
         """Clean up this handler."""
-        for linked_list in self._waiting_dict.values():
-            linked_list.cancel()
-        # All cancelled linked lists should have callbacks to removed itself
-        # from the waiting dict. However, these callbacks are scheduled in
-        # an event loop, so we don't check them now.
-
-    def _unregister_callback(self, fut):
-        del self._waiting_dict[fut.object_id]
-
-    def _complete_future(self, fut):
-        obj = self._worker.get_objects([ray.ObjectID(
-            fut.object_id.binary())])[0]
-        fut.set_result(obj)
+        for futures in self._waiting_dict.values():
+            for fut in futures:
+                fut.cancel()
+
+    def _complete_future(self, ray_object_id):
+        logger.debug(
+            "Completing plasma futures for object id {}".format(ray_object_id))
+
+        obj = ray.get(ray_object_id)
```

Comment:
This apparently causes an error in some places because: `ray.exceptions.RayError: Using blocking ray.get inside async actor`
Should I switch to self._worker.get_objects([ray_object_id])?

Reply:
Yes